### DISCUSSION TITLE
March 21st 2024 Community Call - 7pm PST

### DISCUSSION THREAD

**Discussion by StevenBucher98** [PowerShell/PowerShell#21043] [Community Call]
This call will be used to accommodate those in Asia and Australian time zones and will be hosted at **7pm PST.**

You can join our Teams meeting at https://aka.ms/JoinPSCall

Agenda:

- SSH Inbox @maertendMSFT 
- PSResourceGet Updates @SydneyhSmith 
- PS Gallery Updates @SydneyhSmith 
- VSCode Updates @SydneyhSmith 
- DSC v3 Update @mgreenegit  
- PSSA Release @JamesWTruher 
- PSRL Release @StevenBucher98 
- Doc Updates @sdwheeler

**Comment by jborean93** [comment]
If all goes well I should be able to do a demo on [RemoteForge](https://github.com/jborean93/RemoteForge) and show what it is for and how it all works.

**Comment by ThomasNieto** [comment]
Semantic versioning: https://github.com/PowerShell/PowerShell/issues/2983

* The WG decision does not have any context, address any issues, or consider any alternate solutions.
	As such indicates Semantic Versioning is not supported in PowerShell and by extension PSResourceGet and should be removed

* The scope of work is allowing Semantic versioning in module cmdlets
	which could be accomplished by adding a (65KB) dependency on NuGet.Versioning 

* Here are some of the issues when PSResourceGet supports Semantic Versioning while the engine does not
	Issue 1# - Prerelease binary modules cannot be updated to stable.
		The largest impact of this is PSReadLine as users who install the prerelease version will have to under go workarounds to install the stable version; thus hurting testing adoption

	Issue 2# - Prerelease and stable versions cannot be installed at the same time

	Issue 3# - Formatter displays a fake Prerelease property causing user confusion

* PowerShell should either fully support Semantic versioning in the engine or remove it completely and in PSResourceGet

* My concern about WG verdicts and lack of transparency could be addressed by
	WG meetings being recorded, having meeting minutes, and a publicly accessible vote with accompanying concurring or dissenting opinions.

	This would help users know their issue has been earnestly considered 
	
* I want the PowerShell committee to review the future of Semantic Versioning as PSResourceGet should not support a versioning scheme that PowerShell cannot

**Comment by o-l-a-v** [comment]
Are there any future plans for PowerShell Gallery other than keeping it "as is"?

* Migrate to Nuget v3, or move to something else, like Azure Container Registry (ACR)?
* Use some Nuget v3 shim, like the one @JustinGrote made with `pwsh.gallery`?

Related:

* https://github.com/PowerShell/PowerShellGallery/issues/218

**Reply by o-l-a-v** [reply]
I couldn't join the live call, watched the YT video now. Thanks for the insight and explaination @SydneyhSmith, it all made great sense. 😊

**Comment by NoralK** [comment]
What type of timeframe is PlatyPS v2?  Will the current version get support of Components?

**Comment by kilasuit** [comment]
Here is the [March 2024 Community Call on youtube](https://www.youtube.com/watch?v=YAEOf-rBMLA)
